docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,480 @@
+
+## Why do we want a graph?
+
+There are two use cases I'm currently considering for these compatibility metadata:
+
+- **image selection**: is the primary use case. At face value a graph might not seem valuable, because an algorithm could very simply (in a brute force manner) match a set of attributes to a desired set. And to be clear, that use case is not out of scope - the metadata attributes are provided in the compatibility spec in an easy to read and parse, flat format. However, given that we have these terms map into a graph defined by the schema, we can optimize the image selection process by pruning off entire subtrees (that are impossible, for example) before we even consider more detailed metadata. A scaled image selection task would be more computationally feasible than if using a brute force matching approach. Such an example might be to ask for higher level queries across all images in a registry.
+
+- **scheduling**: Schedulers can often represent resources as graphs (e.g., Flux Framework), and this community would want to be able to easily map one or more compatibility schemas into their scheduler graphs.
+
+The design suggested in this proposal addresses both use cases above.
+
+### Benefits
+
+Several benefits to this design include:
+
+- Parsing (for scheduling or image selection) means we can generate one graph (and don't need to combine graphs)
+- We don't need to handle cases of "single node" graphs (metadata attributes that are nodes but don't have edges)
+- We don't need to keep around an extra jsonschema to validate a custom format, we use a common, community standard (JGF)
+- Simple cases of image selection do not need to consider the graph and can use key value pairs provided in the compatibility spec
+- Changes to the graph (nodes or edges) are in one place (one source of truth) and versioned, so any upstream change does not require updating artifacts.
+
+## Definitions
+
+- A **Compatibility Interest Group** (CIG) has a vested interest in defining compatibility.
+  - Defines namespace for their community metadata
+  - Maintains a schema "compspec.json" in a web-accessible location for their community metadata
+  - Optionally maintains plugins or associated tooling
+- A **Compatibility Schema** is a set of namespaced metadata that describes a compatibiilty interest group's space of expertise.
+  - Maintained by a compatibility interest group
+  - Metadata is described in a graph format, [Json Graph Format](https://github.com/jsongraph/json-graph-specification), version 2.0, with edges and nodes required.
+  - The graph id is the namespace or label for their community metadata (e.g,. `supercontainers.org` or `archspec.io`)
+  - Every node must be represented in the graph (defined in edges), and there should be common validation for this.
+  - Each nodes is a specific metadata attribute (e.g., `cpu` and `cpu.family`)
+  - Each edge is a relationship between two nodes (e.g., `cpu` contains -> `cpu.family`.
```

Comment:
What about attributes from different domains? Are they allowed to build relation?

Reply:
Likely that would be possible, but an "out of tree" use case. For example, maybe I want to describe power attributes for physical resources in a graph scheduler, and they are under different namespaces. The compatibility spec would include the metadata that I need, I would map it to some namespace of labels, and then add a few edges to the graph that are relevant for my specific cluster.